platform/linux-generic/odp_packet.c

```diff
@@ -1780,26 +1778,28 @@ int _odp_packet_copy_md_to_packet(odp_packet_t srcpkt, odp_packet_t dstpkt)
 {
 	odp_packet_hdr_t *srchdr = packet_hdr(srcpkt);
 	odp_packet_hdr_t *dsthdr = packet_hdr(dstpkt);
+	pool_t *src_pool = srchdr->buf_hdr.pool_ptr;
+	pool_t *dst_pool = dsthdr->buf_hdr.pool_ptr;
+	uint32_t src_uarea_size = src_pool->params.pkt.uarea_size;
+	uint32_t dst_uarea_size = dst_pool->params.pkt.uarea_size;
 
 	dsthdr->input = srchdr->input;
 	dsthdr->dst_queue = srchdr->dst_queue;
 	dsthdr->buf_hdr.buf_u64 = srchdr->buf_hdr.buf_u64;
 	if (dsthdr->buf_hdr.uarea_addr != NULL &&
-	    srchdr->buf_hdr.uarea_addr != NULL)
-		memcpy(dsthdr->buf_hdr.uarea_addr,
-		       srchdr->buf_hdr.uarea_addr,
-		       dsthdr->buf_hdr.uarea_size <=
-		       srchdr->buf_hdr.uarea_size ?
-		       dsthdr->buf_hdr.uarea_size :
-		       srchdr->buf_hdr.uarea_size);
+	    srchdr->buf_hdr.uarea_addr != NULL) {
+		memcpy(dsthdr->buf_hdr.uarea_addr, srchdr->buf_hdr.uarea_addr,
```

Comment:
odp_memcpy() might be good fit here.

Reply:
odp_memcpy() is just a wrapper for memcpy() and I would avoid using api functions in the implementation unless there is a clear benefit.

Reply:
ok. I just thought maybe some other implementations already speedup this call and we can get this benefits for free. But I don't might if it will be simple memcpy().